Support removing mailboxes for Professional Email

olaseni opened the pull request:
#### Changes proposed in this Pull Request

* Adds a routine to derive email addresses from the mailbox object
* Writes a method signature for the delete mailbox API 
* Adds a "Delete Mailbox" menu item to remove Professional Email mailbox items
* Wires up the menu item to invoke the delete mailbox API when clicked
* Displays success/error messaging upon response from the API

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Run this branch locally or via the [live branch](https://calypso.live/?branch=add/remove-mailbox-button-for-professional-email).
* Add multiple Professional Email mailboxes for a domain you own.
* Navigate to the email plan page for said domain.
* To observe track events consult the FG Guide at PCYsg-cae-p2
* For each of the mailboxes, open the action/kebab menu for the email.
* Verify that an option exists in the menu to "Remove Mailbox".
* Click on the "Remove Mailbox" option.
* Verify that we triggered a `calypso_email_management_titan_remove_mailbox_click` Tracks event.
* Verify that a confirmation dialog is shown to you
* Verify that you see a success notice appear.
* Verify that the page is updated and no longer shows the removed mailbox.
* Verify that when the last mailbox is removed the email plan page displays "No mailboxes"

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

#### Screenshot

<img width="1140" alt="Screenshot 2021-06-26 at 12 39 52 AM" src="https://user-images.githubusercontent.com/277661/123494287-319e9380-d617-11eb-869d-e1028e4a9efa.png">

a8ci18n commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -91,6 +136,17 @@ const getTitanMenuItems = ( { email, translate } ) => {
+				removeTitanMailbox( dispatch, mailbox, onAccountChanged, translate );
+			},
+			key: `remove_mailbox:${ mailbox.mailbox }`,
+			title: translate( 'Remove Mailbox', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Remove mailbox' )` *ES Score: 13*

stephanethomas commented on client/lib/emails/get-mailbox-address.js:
```diff
@@ -0,0 +1,14 @@
+	if ( mailbox && domain ) {
+		return `${ mailbox }@${ domain }`;
+	}
+	return '';
```
Could we add a blank line above please 🙂?

stephanethomas commented on client/lib/wpcom-undocumented/lib/undocumented.js:
```diff
@@ -800,10 +800,10 @@ Undocumented.prototype.getTitanControlPanelIframeURL = function ( emailAccountId
 
 /**
- * Checks the availability of a mailbox
+ * Checks the availability of a Professional Email(i.e. Titan) mailbox
```
A white space is missing before the opening parenthesis.

stephanethomas commented on client/lib/emails/get-mailbox-address.js:
```diff
@@ -0,0 +1,14 @@
+ * @param {string} Mailbox.domain The domain for the mailbox.
+ * @returns {string} Returns the email address formed from the mailbox object.
+ */
+export function getMailboxAddress( { mailbox, domain } ) {
```
I think this function should be named `getEmail()` or `getEmailAddress()`.

olaseni replied:
I thought this initially as well but I felt it was too generic and may be prone to clashing. I will acquiesce for now and we'll probably handle it if it surfaces in the future.

stephanethomas commented on client/lib/wpcom-undocumented/lib/undocumented.js:
```diff
@@ -818,6 +818,28 @@ Undocumented.prototype.getTitanMailboxAvailability = function ( domain, mailbox,
+		fn
+	);
+};
+
```
As mentioned in p4TIVU-9JT-p2, we are moving away from `wpcom-undocumented`. Could we revisit those changes to use React Query instead? Team Calypso has actually shared some tips not so long ago in p4TIVU-9GP-p2. @AllTerrainDeveloper also started using it in https://github.com/Automattic/wp-calypso/pull/54070.

saygunnyc replied:
Thank you for the ping, @stephanethomas !

> As an aside, I personally find the confirmation message a bit alarming (which in some ways may probably be a good thing).

I think it might make sense to be alarming as they are about to lose their mailbox and data esp if they reached here in error.  Aside from that being alarming, I think there's an opportunity to communicate what the user should expect from the removal a bit in detail here. Sharing a couple of things that came up to my mind,

- When they should expect the mailbox to be removed? Instantly or at the end of the billing cycle?
- What if they want to export their existing Inbox? Is there a way to have a backup? If yes how to do it?

I also don't know the answer to these questions. 😃  Let me know your thoughts!

olaseni replied:
In answer to your questions @saygunnyc 
- The mailbox is removed instantly so they have no access to it. However the data is only deleted 30 days after they removed the mailbox. Within that window, it can only be re-activated manually
- The only way to backup/export their mailbox data right now is to use a third-party mail client.

olaseni replied:
FYI @stephanethomas , I have migrated to react-query

stephanethomas commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -175,19 +236,92 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
+			},
+			message: (
+				<div>
+					<h3>Remove mailbox</h3>
```
This string should be localized.

stephanethomas commented on client/my-sites/email/email-management/home/email-plan.jsx:
```diff
@@ -116,19 +117,25 @@ class EmailPlan extends React.Component {
+	onAccountChanged = () => {
+		this.setState( { hasLoadedEmailAccounts: false, isReloadingEmailAccounts: true } );
+	};
+
```
It looks like we would not need this `onAccountChanged` handler (and all the corresponding props) with React Query (see [this documentation](https://react-query.tanstack.com/guides/invalidations-from-mutations) regarding refetching data after a mutation).

olaseni replied:
Gotcha! Migrated.

stephanethomas commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -175,19 +236,92 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
+			message: (
+				<div>
+					<h3>Remove mailbox</h3>
+					<h4> { emailAddress } </h4>
```
Having the email address on its own line seems a bit strange to me. Maybe we could move it to the title of the modal:

> Remove contact@example.com

Or include it in the copy itself:

>  Are you sure you want to remove contact@example.com? All your data will be deleted!

olaseni replied:
I actually think it stands out more i.e. grabs more attention, and it makes an accommodation for multiple mailboxes when that is eventually supported.

stephanethomas commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -207,40 +341,59 @@ const EmailMailboxActionMenu = ( { account, domain, mailbox } ) => {
-		</EllipsisMenu>
+		<>
+			<RemoveMailboxConfirmationDialog
+				translate={ translate }
```
There is no need to pass `translate`, we can use the `useTranslate()` hook in `RemoveMailboxConfirmationDialog` instead.

stephanethomas commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -175,19 +236,92 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
+} ) => {
+	const onClose = ( action ) => {
+		closeDialog?.();
+		if ( 'remove' === action ) {
```
Let's add another blank line here.

stephanethomas commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -57,15 +61,55 @@ const removeEmailForwardMailbox = ( { dispatch, mailbox } ) => {
+		.then(
+			( response ) => {
+				if ( response.status === 202 ) {
+					onAccountChanged?.();
```
I've never realized that we could use this new syntax for optional callbacks, nice.

stephanethomas commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -207,40 +341,59 @@ const EmailMailboxActionMenu = ( { account, domain, mailbox } ) => {
+			>
+				{ removeMailboxConfirmationItems.message }
+			</RemoveMailboxConfirmationDialog>
+			<EllipsisMenu position="bottom" className="email-mailbox-action-menu__main">
```
And another blank line.

stephanethomas replied:
Looks like we are still missing that blank line 😉.

stephanethomas commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -175,19 +236,92 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
+			),
+		};
+	}
+	return { action: () => {}, message: null };
```
Blank line 🙂.

stephanethomas reviewed (changes requested):
While I was able to delete a mailbox successfully, I have a concern about the technical approach taken which doesn't follow the newest recommendations regarding managing data. @AllTerrainDeveloper is currently working on updating the `Email Plan` page to use React Query (see 1200182182542585-as-1200213040389307), we could take advantage of this.

As an aside, I personally find the confirmation message a bit alarming (which in some ways may probably be a good thing). Let's ping @saygunnyc, and see what he thinks of it:

<img width="627" alt="screenshot" src="https://user-images.githubusercontent.com/594356/123665675-caf8c000-d838-11eb-9480-6748442d1548.png">

stephanethomas commented on client/lib/emails/index.js:
```diff
@@ -1,4 +1,5 @@
 export { getEmailForwardAddress } from './get-email-forward-address';
+export { getEmailAddress } from './get-email-address';
```
This line should be first if we want to keep this list of imports ordered.

stephanethomas commented on client/lib/wpcom-undocumented/lib/undocumented.js:
```diff
@@ -818,6 +818,28 @@ Undocumented.prototype.getTitanMailboxAvailability = function ( domain, mailbox,
 };
 
+/**
+ * Deletes a mailbox from a Professional Email(i.e. Titan) account
```
A white space is missing before the opening parenthesis.

stephanethomas commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -175,19 +236,93 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
+const RemoveMailboxConfirmationDialog = ( {
+	children,
+	closeDialog,
+	confirmationAction,
```
That's minor but since that's an event handler `confirmAction` may be a more appropriate name (i.e. it should start with a verb like `closeDialog`).

stephanethomas commented on client/my-sites/email/email-management/home/email-plan.jsx:
```diff
@@ -141,19 +142,25 @@ class EmailPlan extends React.Component {
 					this.setState( {
 						isLoadingEmailAccounts: false,
 						hasLoadedEmailAccounts: true,
+						isReloadingEmailAccounts: false,
```
Do we really need that new flag? We will have to remove it when migrating to React Query. If it's just to avoid displaying placeholders when reloading the list of email accounts, then maybe we could use/pass `hasLoadedEmailAccounts`:

```javascript
<EmailPlanMailboxesList
  account={ this.getAccount() }
  domain={ domain }
  isLoadingEmails={ ! hasLoadedEmailAccounts }
  mailboxes={ this.getMailboxes() }
  onAccountChanged={ this.onAccountChanged }
/>
```

stephanethomas commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -91,6 +135,22 @@ const getTitanMenuItems = ( { email, translate } ) => {
+			},
+			key: `remove_mailbox:${ mailbox.mailbox }`,
+			title: translate( 'Remove mailbox', {
+				comment: 'Remove a mailbox',
```
Do you think this comment would be useful to translators?

stephanethomas reviewed (approved):
I was able to remove a mailbox successfully, although the list of mailboxes is not refreshed in that case:

<img width="632" alt="screenshot" src="https://user-images.githubusercontent.com/594356/123975622-c7427600-d9bd-11eb-9e5d-cd81eeb013fe.png">

This is a similar issue than was I've mentioned in https://github.com/Automattic/wp-calypso/pull/54133#pullrequestreview-696097153. It's probably better to address it when we migrate this code to React Query (although `onAccountChanged` doesn't seem to work so maybe we want to remove it for now).

olaseni commented:
> I was able to remove a mailbox successfully, although the list of mailboxes is not refreshed in that case:
> 
> <img alt="screenshot" width="632" src="https://user-images.githubusercontent.com/594356/123975622-c7427600-d9bd-11eb-9e5d-cd81eeb013fe.png">
> 
> This is a similar issue than was I've mentioned in [#54133 (review)](https://github.com/Automattic/wp-calypso/pull/54133#pullrequestreview-696097153). It's probably better to address it when we migrate this code to React Query (although `onAccountChanged` doesn't seem to work so maybe we want to remove it for now).

I'm not able to reproduce this. This screenshots references email forwards @stephanethomas

stephanethomas commented:
@olaseni: I forgot to mention that I've also spotted a wrong import:

```
ERROR in ./my-sites/email/email-management/home/email-mailbox-action-menu.jsx 250:23-38
export 'getEmailAddress' (imported as 'getEmailAddress') was not found in 'calypso/lib/emails' (possible exports: getEmailForwardAddress, hasGoogleAccountTOSWarning, hasUnusedMailboxWarning, hasUnverifiedEmailForward, isEmailForward, isEmailForwardAccount, isEmailForwardVerified, isEmailUserAdmin, isGoogleEmailAccount, isTitanMailAccount)
 @ ./my-sites/email/email-management/home/email-plan-mailboxes-list.jsx 15:0-108 131:41-63
 @ ./my-sites/email/email-management/home/email-plan.jsx 20:0-108 342:41-63
 @ ./my-sites/email/email-management/email-home.jsx 21:0-80 75:76-85
 @ ./my-sites/email/controller.js 11:0-85 74:127-146
 @ ./my-sites/email/index.js 11:0-38 27:34-60 32:34-74 36:34-78 40:34-80 44:34-83 48:34-75 53:15-66 57:34-70
 @ ./sections.js 312:16-314:27
 @ ./sections-middleware.js 20:0-34 21:16-24 115:24-32
 @ ./boot/common.js 23:0-58 276:2-13
 @ ./boot/app.js 7:0-35 15:2-9
```

stephanethomas commented:
> I'm not able to reproduce this. This screenshots references email forwards @stephanethomas

Oops, I've posted the wrong file. Here is the right one:

<img width="632" alt="screenshot" src="https://user-images.githubusercontent.com/594356/123977202-14731780-d9bf-11eb-9e45-0156e2500937.png">

stephanethomas commented:
> I'm not able to reproduce this. 

FWIW, I've noticed that issue on `calypso.live` (I didn't test the changes locally).

olaseni committed:
Remove unneeded undocumented.js function `getEmailAccountsForSiteAndDomain`

olaseni committed:
Add mutation

olaseni commented:
> @olaseni: I forgot to mention that I've also spotted a wrong import:
> 
> ```
> ERROR in ./my-sites/email/email-management/home/email-mailbox-action-menu.jsx 250:23-38
> export 'getEmailAddress' (imported as 'getEmailAddress') was not found in 'calypso/lib/emails' (possible exports: getEmailForwardAddress, hasGoogleAccountTOSWarning, hasUnusedMailboxWarning, hasUnverifiedEmailForward, isEmailForward, isEmailForwardAccount, isEmailForwardVerified, isEmailUserAdmin, isGoogleEmailAccount, isTitanMailAccount)
>  @ ./my-sites/email/email-management/home/email-plan-mailboxes-list.jsx 15:0-108 131:41-63
>  @ ./my-sites/email/email-management/home/email-plan.jsx 20:0-108 342:41-63
>  @ ./my-sites/email/email-management/email-home.jsx 21:0-80 75:76-85
>  @ ./my-sites/email/controller.js 11:0-85 74:127-146
>  @ ./my-sites/email/index.js 11:0-38 27:34-60 32:34-74 36:34-78 40:34-80 44:34-83 48:34-75 53:15-66 57:34-70
>  @ ./sections.js 312:16-314:27
>  @ ./sections-middleware.js 20:0-34 21:16-24 115:24-32
>  @ ./boot/common.js 23:0-58 276:2-13
>  @ ./boot/app.js 7:0-35 15:2-9
> ```

I can't reproduce this. However, since the migration to react-query, this issue doesn't exist anymore.

stephanethomas commented on client/data/emails/use-remove-titan-mailbox-mutation.js:
```diff
@@ -0,0 +1,58 @@
+	mutationOptions.onSettled = ( data, error, variables, context ) => {
+		suppliedOnSettled?.( data, error, variables, context );
+
+		queryClient.invalidateQueries( {
+			predicate: ( { queryKey: [ prefix, , domain ] = [] } ) =>
+				prefix === queryKeyPrefix && domain === domainName,
+		} );
```
As far I can see, we are no longer taking the site identifier in consideration when invalidating data. Wouldn't that be a problem? Why do we need to define a custom predicate? Can't we reuse our cache key like this instead?

```suggestion
		queryClient.invalidateQueries( getCacheKey( siteId, domainName ) );
```

olaseni replied:
We don't need to pass in the siteId just to invalidate the cache. React-query is flexible enough to work with a predicate.

stephanethomas replied:
I understand we can pass a predicate. The thing that I don't is that we use a cache key when are fetching data that is different from when we are mutating that same data.

olaseni replied:
It's more accurately called a `queryKey` not a cache key. We can supply a portion, or prefix of the key to get the runtime to invalidate all matching queries.

stephanethomas commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -175,19 +197,117 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
+				},
+			}
+		),
+		{ duration: 7000 }
```
Is that an arbitrary value? Is there a constant defined somewhere in Calypso that we could use instead?

olaseni replied:
None that I know of

stephanethomas commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -175,19 +197,117 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
+				strong: <strong />,
+			},
+		} ),
+		{ duration: 5000 }
```
Same thing about this hardcoded value.

stephanethomas commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -175,19 +197,117 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
+
+	const { removeTitanMailbox } = useRemoveTitanMailboxMutation( mailbox.domain, mailbox.mailbox, {
+		onSettled: ( data ) => {
+			if ( data?.status === 202 ) {
+				dispatch( successMessage );
+				return;
+			}
+			dispatch( errorMessage );
```
Let's add blank lines:

```suggestion
			if ( data?.status === 202 ) {
				dispatch( successMessage );

				return;
			}

			dispatch( errorMessage );
```

stephanethomas commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -175,19 +197,117 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
-	const email = `${ mailbox.mailbox }@${ mailbox.domain }`;
+	const [ removeTitanMailboxDialogVisible, setRemoveTitanMailboxDialogVisible ] = useState( false );
+
+	const domainHasTitanMailWithUs = hasTitanMailWithUs( domain );
```
Do we really need that variable? In that particular case, I don't think it brings much value. The list of conditions below was easier to read with all the `hasSomething()` calls.

olaseni replied:
On the contrary, I think we do need the variable for simplicity and to ensure that we are not repeating the same calls within the same block.

stephanethomas commented on client/data/emails/use-remove-titan-mailbox-mutation.js:
```diff
@@ -0,0 +1,58 @@
+
+	const queryClient = useQueryClient();
+
+	mutationOptions.onSettled = ( data, error, variables, context ) => {
```
Why do we need to define these mutations options? I'm actually not sure what `onSettled` means. It doesn't seem necessary looking at the other places in Calypso that define a mutation for React Query ([an example](https://github.com/Automattic/wp-calypso/blob/d64f25956d54275c213f948b3e06180267d73482/client/data/users/use-update-user-mutation.js#L13)), and which actually look simpler.

olaseni replied:
Please take a look at the definition of the `onSettled` feature. This utilisation is not so different from other examples in Calypso, even though the solutions are not necessarily similar.

stephanethomas commented on client/data/emails/use-remove-titan-mailbox-mutation.js:
```diff
@@ -0,0 +1,58 @@
+ * @param {object} mutationOptions Mutation options passed on to `useMutation`
+ * @returns {{ data, error, isLoading: boolean, removeTitanMailbox: Function, ...}} Returns various parameters piped from `useMutation`
+ */
+export function useRemoveTitanMailboxMutation( domainName, mailboxName, mutationOptions = null ) {
```
Why do we have that `mutationOptions` parameter? We never pass such parameter when we call that function, so we could probably remove it to keep our code simple (we can still add it back whenever it is really needed).

olaseni replied:
Please take another look. The options are being used.

stephanethomas reviewed (changes requested):
The way the React Query mutator was implemented looks a bit convoluted to me. It's true that I don't have any experience with that library but still I'm surprised to see how much we diverged from other mutators already implemented in Calypso.

I'm also afraid that the list is still not refreshed when a mailbox is removed, which is a bit confusing:

![video](https://user-images.githubusercontent.com/594356/124130805-72673400-da7f-11eb-8fbc-f4c9aaa6f2dc.gif)

olaseni committed:
Add selectedId selector to hook

olaseni commented:
> The way the React Query mutator was implemented looks a bit convoluted to me. It's true that I don't have any experience with that library but still I'm surprised to see how much we diverged from other mutators already implemented in Calypso.
> 

I'm afraid I don't see the divergence you purport. I don't think there should be a problem with a unique solution, regardless.

> I'm also afraid that the list is still not refreshed when a mailbox is removed, which is a bit confusing:
> 

The list has been refreshed on my trials on invalidating the cache. However I have now added the siteId as part of the query key and added optimistic updates to improve the refresh state.

olaseni committed:
Monitor number of mailboxes in mutation

olaseni committed:
Remove debug tools

AllTerrainDeveloper reviewed (approved):
I liked a lot how you have implemented the delete mutation in `client/data/emails/use-remove-titan-mailbox-mutation.js`! Well done! 🏆

daledupreez commented on client/data/emails/use-remove-titan-mailbox-mutation.js:
```diff
@@ -0,0 +1,93 @@
+	// This is called for both success and error statuses.
+	// Invoke any supplied `onSettled` callbacks here. This is so we can do things like invalidate queries etc
+	mutationOptions.onSettled = ( data, error, variables, context ) => {
+		suppliedOnSettled?.( data, error, variables, context );
```
Nit: given that we're defaulting in `noop` for `suppliedOnSettled`, I am not sure the `?.` operator is needed here.

olaseni replied:
I think this one is still good as it ensures that we can call a variable as a function.

daledupreez commented on client/data/emails/use-remove-titan-mailbox-mutation.js:
```diff
@@ -0,0 +1,93 @@
+		queryClient.invalidateQueries( queryKey ).then( () => {
+			const numberOfMailboxes = getNumberOfMailboxes( queryClient, queryKey );
+
+			// Determine if we need to schedule another invalidation since the removal job is not synchronous
```
This comment could be clearer, as it describes what happens after the `if` condition. How about the following?
```suggestion
			// Determine if we already have updated data, since the removal job is not synchronous
			// If we don't have the updated data yet, queue up another invalidation
```

daledupreez commented on client/data/emails/use-remove-titan-mailbox-mutation.js:
```diff
@@ -0,0 +1,93 @@
+
+			setTimeout( () => {
+				queryClient.invalidateQueries( queryKey );
+			}, 10000 );
```
Two non-blocking comments here:
1. It may be worth pushing this value into a constant at the top of the file to make this more visible, e.g. `const INVALIDATION_DELAY = 10000;`
2. Why 10 seconds? That feels really long and likely to trigger unexpected behaviour given the delay. I'd think something like 5 seconds would be enough.

daledupreez commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -61,11 +65,14 @@ const removeEmailForwardMailbox = ( { dispatch, mailbox } ) => {
  * @returns Array of menu items
  */
-const getTitanMenuItems = ( { email, translate } ) => {
+const getTitanMenuItems = ( { showRemoveMailboxDialog, mailbox, translate } ) => {
```
Nit: these arguments aren't in alphabetical order.
```suggestion
const getTitanMenuItems = ( { mailbox, showRemoveMailboxDialog, translate } ) => {
```

daledupreez commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -175,19 +197,124 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
+		translate(
+			'There was an error removing {{strong}}%(emailAddress)s{{/strong}} from your account',
+			{
+				comment: '%(emailAddress)s is the email address associated to the mailbox being deleted',
```
Nit: it might be simpler to express this as:
```suggestion
				comment: '%(emailAddress)s is the email address for the mailbox being deleted',
```

daledupreez commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -175,19 +197,124 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
+		translate(
+			'{{strong}}%(emailAddress)s{{/strong}} has been scheduled for removal from your account',
+			{
+				comment: '%(emailAddress)s is the email address associated to the mailbox being deleted',
```
As above, nit with "associated to" => "for"

daledupreez commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -175,19 +197,124 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
+						'Are you sure you want to remove {{strong}}%(emailAddress)s{{/strong}}? All your data will be deleted!',
+						{
+							comment:
+								'%(emailAddress)s is the email address associated to the mailbox being deleted',
```
Again, "associated to" => "for"

daledupreez commented on client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx:
```diff
@@ -175,19 +197,124 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
+				<h3> { translate( 'Remove mailbox' ) } </h3>
+				<p>
+					{ translate(
+						'Are you sure you want to remove {{strong}}%(emailAddress)s{{/strong}}? All your data will be deleted!',
```
Should we be specific about what "data" is involved? Might it be clearer to say something like:
> All your emails, calendar events, and contacts will be deleted!

This is non-blocking.

daledupreez reviewed (approved):
Overall, this LGTM! I would like the 10 second delay for the invalidation retry to be reduced, or for the reasoning to be documented, but that doesn't necessarily block shipping this.

One notable issue that we should fix in a follow-up PR is that the delete icon is black, while all the icons and text are a grey color. I am honestly not sure of the grey color for the icons. @saygunnyc, can you confirm what the fill shade should be?

olaseni committed:
Lexical changes

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6128039

Thank you @olaseni for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6128039).
